pkg/queue/health/probe.go

```diff
@@ -91,12 +91,8 @@ var transport = func() *http.Transport {
 	return t
 }()
 
-func getURL(config HTTPProbeConfigOptions) url.URL {
-	return url.URL{
-		Scheme: string(config.Scheme),
-		Host:   net.JoinHostPort(config.Host, config.Port.String()),
-		Path:   config.Path,
-	}
+func getURL(config HTTPProbeConfigOptions) (*url.URL, error) {
+	return url.Parse(string(config.Scheme) + "://" + net.JoinHostPort(config.Host, config.Port.String()) + config.Path)
```

Comment:
For clarification, given the path  `/mypath?foo=bar` this is equal to constructing a URL with:
```
		Path:    "/mypath",
		RawQuery: "foo=bar",
```
u.String() method indeed uses an escaped path only so does not seem to have a way to avoid it. Also I think this is the proper way to do it as we want some parsing to happen instead of manually creating the URL. However, there are issues with parsing https://github.com/golang/go/issues/40894.